python/tests/io/test_hdf5.py

```diff
@@ -35,35 +71,44 @@ def test_data_array_all_units_supported():
 
 
 def test_data_array_1d():
-    a = sc.DataArray(data=x,
-                     coords={
-                         'x': x,
-                         'x2': 2.0 * x
-                     },
-                     masks={
-                         'mask1': sc.less(x, 1.5 * sc.units.m),
-                         'mask2': sc.less(x, 2.5 * sc.units.m)
-                     },
-                     unaligned_coords={
-                         'attr1': x,
-                         'attr2': 1.2 * sc.units.K
-                     })
-    check_roundtrip(a)
+    check_roundtrip(array_1d)
 
 
 def test_data_array_2d():
-    a = sc.DataArray(data=xy,
-                     coords={
-                         'x': x,
-                         'y': y,
-                         'x2': 2.0 * x
-                     },
-                     masks={
-                         'mask1': sc.less(x, 1.5 * sc.units.m),
-                         'mask2': sc.less(xy, 0.5 * sc.units.kg)
-                     },
-                     unaligned_coords={
-                         'attr1': xy,
-                         'attr2': 1.2 * sc.units.K
-                     })
+    check_roundtrip(array_2d)
+
+
+def test_data_array_dtype_DataArray():
+    a = sc.DataArray(data=x)
+    a.coords['scalar'] = sc.Variable(value=a)
```

Comment:
In a similar manner. How about Variables storing a `Dataset` as value?

Reply:
Good point, added.